dhis-2/dhis-services/dhis-service-program-rule/src/main/java/org/hisp/dhis/programrule/hibernate/HibernateProgramRuleStore.java

```diff
@@ -62,7 +65,20 @@ public ProgramRule getByName( String name, Program program )
             .addPredicate( root -> builder.equal( root.get( "name" ), name ) )
             .addPredicate( root -> builder.equal( root.get( "program" ), program ) ) );
     }
-    
+
+    @Override
+    @SuppressWarnings("unchecked")
+    public List<ProgramRule> getImplementableProgramRules( Program program, Set<ProgramRuleActionType> types )
+    {
+        Query q = getQuery("FROM ProgramRule pr " +
+            "JOIN FETCH pr.programRuleActions pra  WHERE pr.program = :programId AND pra.programRuleActionType IN (:implementableTypes)" );
+
+        q.setParameter( "programId", program );
+        q.setParameter( "implementableTypes", types );
+
+        return (List<ProgramRule>) q.getResultList();
+    }
+
```

Comment:
We can remove @SuppressWarnings("unchecked") with this 
```suggestion
return getQuery("FROM ProgramRule pr " +
             "JOIN FETCH pr.programRuleActions pra  WHERE pr.program = :programId AND pra.programRuleActionType IN (:implementableTypes)" )
         .setParameter( "programId", program );
         .setParameter( "implementableTypes", types )
         .getResultList();
```